src/main.cpp

```diff
@@ -26,14 +26,37 @@ bool isInRecoveryMode() {
   }
   saveSettings();
   DPRINTF("Wrote QPC = %d\n", Settings->quick_power_cycle_count);
+  // making sure the board can't go into a locked state
+  Settings->board_locked = false;
   return recovery;
+  return false;
+}
+
+bool isLocked() {
+  // no password set or board isn't armed, won't lock
+  if (strlen(Settings->ap_self_password) == 0 || !Settings->board_lock_armed ||
+      !Settings->board_locked) {
```

Comment:
Thinking more about this, !Settings->board_locked is unnecessary here. If the lock isn't armed, we shouldn't lock.